### PR TITLE
added distanceCounter

### PR DESCRIPTION
**Pull request by Mdominykas**
- Added DistanceCounter - class that calculates distance between current location and given Location
- Distance is displayed in LocationDetails
- Changed CoordX, CoordY to Longitude and Lattitude in Models.Location
- Added radiobutton that allows to display distance in meters or yards (fullfilled delegate requirement)

### CODE REVIEW

**Review by Waltzerizm — APPROVED** [review]
Looks good. Just curious why write a distance calculating function when there is Xamarin.Essentials.Location.CalculateDistance()? For the sake of requirement completion?
Also the word "Latitude" is misspelled everywhere D:

File: Encountify/ViewModels/LocationDetailViewModel.cs [hunk 1]
```diff
@@ -14,6 +15,8 @@ public class LocationDetailViewModel : BaseViewModel
         private double coordX;
```
Comment (MigleMic) [inline Encountify/ViewModels/LocationDetailViewModel.cs:15]: Maybe we should change coordX and coordY here to longitude and latitude as well.

File: Encountify/ViewModels/NewLocationViewModel.cs [hunk 2]
```diff
@@ -74,8 +74,8 @@ private async void OnSave()
             {
                 Name = Name,
                 Description = Description,
-                CoordX = CoordX,
-                CoordY = CoordY,
+                Longitude = CoordX,
+                Lattitude = CoordY,
```
Comment (MigleMic) [inline Encountify/ViewModels/NewLocationViewModel.cs:78]: It would also be easier to read here if it would be Latitude = Latitude, not Latitude = CoordY

**Review by MigleMic — APPROVED** [review]
Works great, I just think that some variables should be changed as I mentioned in comments.

File: Encountify/ViewModels/LocationDetailViewModel.cs (hunk 1 above)
Comment (Mdominykas) [reply Encountify/ViewModels/LocationDetailViewModel.cs]: Forgot to change it

**Comment by Mdominykas** [comment]
> 
> 
> Looks good. Just curious why write a distance calculating function when there is Xamarin.Essentials.Location.CalculateDistance()? For the sake of requirement completion? Also the word "Latitude" is misspelled everywhere D:

Before CalculateDistance seemed to not work (now changed to it) and lattitude now is with correct spelling.
The idea for using delegates is that doing so moves complexity away from place where it is used. Here it means that LocationDetailPage does not care how user wants to view data, it will be presented with same call. First I tried to do different functions for distance (crow flight vs walking), but it did not worked, so I settled with yards vs meters and might try it again later on

**Pull request merged**